Bourreau/script/cbrain_remote_ctl

```diff
@@ -24,113 +24,110 @@
-# to be started manually by users).
-#
+# this bash script is kept simple).
 #
+# -----------------------
 # To start the Rails app:
 #
 #   $0 start -e <environment>
 #
 # which will result in reading the database.yml from stdin, installing
 # it, executing this command
 #
+#   script/puma_wait_wrapper
+#
+# which starts
+#
 #   puma --config config/puma.rb
 #
 # and finally removing the database.yml file. Any database.yml file
 # already present before doing all this will be erased.
 #
-#
+# ----------------------
 # To stop the Rails app:
 #
 #   $0 stop
 #
 # which will result in killing the rails app.
 #
-#
+# -------------------
 # To start a console:
 #
 #   $0 console -e <environment>
 
-# IMPORTANT DEVELOPER NOTICE:
-#
-# This script is written in plain standard Ruby with NO
-# access to Rails and other ActiveSupport helpers. Therefore
-# you can't invoke things like ".present?" and ".blank?" etc.
-# The reason for this is to make this script boot up as
-# quickly as possible.
-
 #########################
 # S U B R O U T I N E S #
 #########################
 
 # Usage
-def usage(basename)
-    puts "Usage: #{basename} start -e <environment>\n" +
-         "       #{basename} stop\n" +
-         "       #{basename} console -e <environment>"
-    Kernel.exit(10)
-end
+function usage {
+  echo "Usage: $0 start -e <environment>"
+  echo "       $0 stop"
+  echo "       $0 console -e <environment>"
+  exit 10
+}
 
 # Fatal error message
-def fatal(message)
-    puts message
-    Kernel.exit(20)
-end
+function fatal {
+  echo "$*"
+  exit 20
+}
 
-# Run-time paths and names of host and program
-basename   = $PROGRAM_NAME.sub(/.*\//,"") # that constant is provided by Ruby
-curhost    = `hostname -s`.strip
-rails_home = __dir__.sub(/\/Bourreau\/script.*$/,"/Bourreau")
+# This program only works if its CWD is the Bourreau's root dir.
+# We verify this by checking for the presence of a particular file
+if ! test -f "app/models/bourreau_worker.rb" ; then
+  fatal "This program '$0' must be started from the Bourreau directory."
+fi
 
-# Make sure we're at the base of the app
-Dir.chdir(rails_home)
+# Run-time paths and names of host and program
+basename=$(basename $0)
+curhost=$(hostname -s)
+rails_home=$PWD
 
 # Rails-specific paths.
 # Alongside the PID file, we maintain the hostname where
 # the Bourreau was previously started.
-db_file    = "config/database.yml"
-pidfile    = "tmp/pids/server.pid"
-pidhost    = "#{pidfile}.hostname"
+db_file="config/database.yml"
+pidfile="tmp/pids/server.pid"
+pidhost="$pidfile.hostname"
 
 # Values from previous 'start' operation
-prevpid    = File.read(pidfile).to_i rescue 0
-prevhost   = (File.read(pidhost) rescue "").strip
-
-# To help diagnose problems, print Ruby's version.
-puts "You are using Ruby #{RUBY_VERSION}"
-if RUBY_VERSION =~ /^1\.[0-8]/
-  puts "Warning! This Ruby version seems to be way too old. Maybe your environment wasn't initialized properly?"
-end
+prevpid=$(cat $pidfile 2>/dev/null)
```

Comment:
not sure what are you doing. Seems to evaluate to empty string

Reply:
It's so the statement works when the file doesn't exist and don't print error messages. `2>` is a redirection of channel #2 (stderr), not stdout (which is channel #1).

Why don't you try it on the command line yourself:

```shell
echo 1234 > pidfile
x=$(cat pidfile 2>/dev/null) # with the file existing
echo "-$x-"
rm pidfile
x=$(cat pidfile 2>/dev/null) # with the file not existing
echo "-$x-"
```